tools/c7n_kube/c7n_kube/server.py

```diff
@@ -0,0 +1,137 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+import json
+import os
+import http.server
+
+from c7n.config import Config
+from c7n.loader import DirectoryLoader
+
+import logging
+
+log = logging.getLogger("c7n_kube.server")
+log.setLevel(logging.DEBUG)
+
+
+HOST = "0.0.0.0"
+
+
+class AdmissionControllerServer(http.server.HTTPServer):
+    """
+    Admission Controller Server
+    """
+
+    def __init__(self, policy_dir, *args, **kwargs):
+        self.policy_dir = policy_dir
+        self.directory_loader = DirectoryLoader(Config.empty())
+        policy_collection = self.directory_loader.load_directory(
+            os.path.abspath(self.policy_dir))
+        self.policy_collection = policy_collection.filter(modes=['k8s-validator'])
+        log.info(f"Loaded {len(self.policy_collection)} policies")
+        super().__init__(*args, **kwargs)
+
+
+class AdmissionControllerHandler(http.server.BaseHTTPRequestHandler):
+    def get_request_body(self):
+        token = self.rfile.read(int(self.headers["Content-length"]))
+        res = token.decode("utf-8")
+        return res
+
+    def do_GET(self):
+        """
+        Returns application/json list of your policies
+        """
+        self.send_response(200)
+        self.end_headers()
+        result = []
+        for p in self.server.policy_collection.policies:
+            result.append(p.data)
+        self.wfile.write(json.dumps(result).encode('utf-8'))
+
+    def do_POST(self):
+        """
+        Entrypoint for kubernetes webhook
+        """
+        req = self.get_request_body()
+        log.info(req)
+        try:
+            req = json.loads(req)
+        except Exception as e:
+            self.send_response(400)
+            self.send_header('Content-Type', 'application/json')
+            self.end_headers()
+            self.wfile.write(json.dumps({"error": str(e)}).encode('utf-8'))
+            return
+
+        results = []
+        failed_policies = []
+        warn_policies = []
+        for p in self.server.policy_collection.policies:
+            result, resources = p.push(req)
```

Comment:
this needs a try/except on it. 500 error.

Reply:
I think we want to catch and respond with a 200 if there's an exception while running the policy with the exceptions in the warnings by default? and have it configurable to fail the admission on policy failure as an alternative